.github/workflows/r.yml

```diff
@@ -32,78 +32,70 @@ on:
       - 'r/**'
 
 jobs:
-
-  ubuntu:
-    name: AMD64 Ubuntu ${{ matrix.ubuntu }} R ${{ matrix.r }}
+  r-hub:
```

Comment:
@kszucs I replaced the ubuntu-r and conda-r GHA jobs with these two. This gets us coverage on the development version of R and using an r-hub image designed to emulate CRAN, as well as one on CentOS, which uncovered a bug in the process of doing this PR. 

The old ubuntu-r and conda-r jobs are being tested nightly by crossbow still.

I then added to crossbow the other 5 new docker images from the build matrix I originally had here.